yt_dlp/extractor/redgifs.py

```diff
@@ -80,15 +75,196 @@ def _real_extract(self, url):
             })
         self._sort_formats(formats)
 
+        webpage_url = f'https://redgifs.com/watch/{video_id}'
+
         return {
             'id': video_id,
-            'title': ' '.join(gif.get('tags') or []) or 'RedGifs',
-            'timestamp': int_or_none(gif.get('createDate')),
-            'uploader': gif.get('userName'),
-            'duration': int_or_none(gif.get('duration')),
-            'view_count': int_or_none(gif.get('views')),
-            'like_count': int_or_none(gif.get('likes')),
-            'categories': gif.get('tags') or [],
+            'title': ' '.join(gif_data.get('tags') or []) or 'RedGifs',
+            'timestamp': int_or_none(gif_data.get('createDate')),
+            'uploader': gif_data.get('userName'),
+            'duration': int_or_none(gif_data.get('duration')),
+            'view_count': int_or_none(gif_data.get('views')),
+            'like_count': int_or_none(gif_data.get('likes')),
+            'categories': gif_data.get('tags') or [],
+            'webpage_url': webpage_url,
```

Comment:
Since one extractor is directly retuning another's result, add `ie_key` and `extractor` as well

Reply:
I don't see those listed in common.py, unless returning a `'_type': 'url'` for use by another extractor? Since they're all returning full results from a base class, is that needed?

Reply:
The issue here is that the search extractor is directly returning entries that would normally be extracted by the video extractor. This is why `webpage_url`, `ie_key` and `extractor` has to be set

Reply:
Ah, fair enough, added those then